### PR TITLE
Add metadata frontmatter for AE glossary

### PR DESCRIPTION
**Pull request by johnblust**
## Description & motivation
Adds meta title and meta description inputs into the AE glossary entries using the `description:` field and

```
<head>
     <title></title>
</head>
```

### CODE REVIEW

File: website/docs/terms/dag.md [hunk 1]
```diff
@@ -1,4 +1,14 @@
-# DAG
+---
+id: dag
+title: DAG
+description: Data extraction is the process by which data is retrieved from multiple sources, often varying in volume and structure.
```
Comment (KiraFuruichi) [inline website/docs/terms/dag.md]: wrong description for DAG

File: website/docs/terms/dag.md [hunk 2]
```diff
@@ -1,4 +1,14 @@
...
+displayText: DAG
+hoverSnippet: Data extraction is the process by which data is retrieved from multiple sources, often varying in volume and structure.
```
Comment (KiraFuruichi) [inline website/docs/terms/dag.md]: wrong hoverSnippet for DAG

File: website/docs/terms/data-extraction.md [hunk 3]
```diff
@@ -1,10 +1,15 @@
 ---
 id: data-extraction
 title: Data extraction
+description: Data extraction is the process by which data is retrieved from multiple sources, often varying in volume and structure.
 displayText: data extraction
 hoverSnippet: Data extraction is the process by which data is retrieved from multiple sources, often varying in volume and structure.
 ---
 
+<head>
+    <title>Data extraction: the first step of the ELT process</title>
```
Comment (KiraFuruichi) [inline website/docs/terms/data-extraction.md]:
```suggestion
    <title>Data extraction: The first step of the ELT process</title>
```

File: website/docs/terms/data-lineage.md [hunk 4]
```diff
@@ -1,10 +1,15 @@
 ---
 id: data-lineage
 title: Data lineage
+description: Data lineage provides a holistic view of how data moves through an organization, where it’s transformed and consumed.
 displayText: data lineage
 hoverSnippet: Data lineage provides a holistic view of how data moves through an organization, where it’s transformed and consumed.
 ---
 
+<head>
+    <title>Data lineage: what it is and how to automate it - dbt Labs</title>
```
Comment (KiraFuruichi) [inline website/docs/terms/data-lineage.md]:
```suggestion
    <title>Data lineage: What it is and how to automate it - dbt Labs</title>
```

File: website/docs/terms/data-warehouse.md [hunk 5]
```diff
@@ -1,10 +1,15 @@
 ---
 id: data-warehouse
 title: Data warehouse
+description: How have data warehouses evolved over the last 40 years? Explore the nuanced changes in use case since Inmon first coined the term.
 displayText: data warehouse
 hoverSnippet: A data warehouse is a data management system used for data storage and computing that allows for analytics activities such as transforming and sharing data.
 ---
 
+<head>
+    <title>Data warehouses in the post-modern data stack - dbt Labs</title>
```
Comment (KiraFuruichi) [inline website/docs/terms/data-warehouse.md]:
```suggestion
    <title>Data warehouses in the modern data stack - dbt Labs</title>
```

File: website/docs/terms/dry.md [hunk 6]
```diff
@@ -1,10 +1,15 @@
 ---
 id: dry
 title: DRY
+description: DRY is a software development principle that stands for “Don’t Repeat Yourself.” Living by this principle means that your aim is to reduce repetitive patterns and code.
 displayText: DRY  
 hoverSnippet: DRY is a software development principle that stands for “Don’t Repeat Yourself.” Living by this principle means that your aim is to reduce repetitive patterns and duplicate code and logic in favor of modular and referenceable code.
 ---
 
+<head>
+  <title>What is DRY? Hint: It makes for great code - dbt Labs</title>
```
Comment (KiraFuruichi) [inline website/docs/terms/dry.md:10]:
```suggestion
  <title>What is DRY code? Hint: It makes for a great dbt project - dbt Labs</title>
```

File: website/docs/terms/reverse-etl.md [hunk 7]
```diff
@@ -1,10 +1,15 @@
 ---
 id: reverse-etl
 title: Reverse ETL
+description:  Reverse ETL is the process of getting your transformed data stored in your data warehouse to end business platforms, such as sales CRMs and ad platforms
```
Comment (KiraFuruichi) [inline website/docs/terms/reverse-etl.md]:
```suggestion
description:  Reverse ETL is the process of getting your transformed data stored in your data warehouse to end business platforms, such as sales CRMs and ad platforms.
```

File: website/docs/terms/reverse-etl.md [hunk 8]
```diff
@@ -1,10 +1,15 @@
...
 displayText: reverse ETL
 hoverSnippet: Reverse ETL is the process of getting your transformed data stored in your data warehouse to end business platforms, such as sales CRMs and ad platforms. 
 ---
 
+<head>
+    <title>Reverse ETL, demystified: What it is in plain english</title>
```
Comment (KiraFuruichi) [inline website/docs/terms/reverse-etl.md:10]:
```suggestion
    <title>Reverse ETL, demystified: What it is in plain english?</title>
```

File: website/docs/terms/subquery.md [hunk 9]
```diff
@@ -1,10 +1,14 @@
 ---
 id: subquery
 title: Subquery in SQL
+description: "A subquery is what the name suggests: a query within another query. The true inception of SQL. Subqueries are often used when you need to process data in several steps."
```
Comment (KiraFuruichi) [inline website/docs/terms/subquery.md:4]: is the `description` field supposed to be in quotes?

File: website/docs/terms/subquery.md [hunk 10]
```diff
@@ -1,10 +1,14 @@
...
 displayText: subquery
 hoverSnippet: A subquery is a query within another query. Subqueries are often used when you need to process data in multiple steps.
 ---
 
+<head>
+  <title>What is a Subquery in SQL and when they are useful - dbt Labs</title>
```
Comment (KiraFuruichi) [inline website/docs/terms/subquery.md]:
```suggestion
  <title>What is a Subquery in SQL and when are they useful? - dbt Labs</title>
```

**Review by KiraFuruichi — CHANGES_REQUESTED** [review]
Left some comments

File: website/docs/terms/subquery.md (hunk 9 above)
Comment (johnblust) [reply website/docs/terms/subquery.md:4]: If there is a semi-colon present, yes. Otherwise, it breaks the frontmatter format

File: website/docs/terms/reverse-etl.md (hunk 8 above)
Comment (johnblust) [reply website/docs/terms/reverse-etl.md:10]: "What it is in plain english" is a statement, and does require a question mark

File: website/docs/terms/dry.md (hunk 6 above)
Comment (johnblust) [reply website/docs/terms/dry.md:10]: I want to stay away from product mention here, it's not relevant to non-dbt users

File: website/docs/terms/data-warehouse.md (hunk 5 above)
Comment (johnblust) [reply website/docs/terms/data-warehouse.md]: Any reason why you don't agree with phrasing here? I'm trying to give the impression that this is a more updated definition then everything else on page one

Comment (KiraFuruichi) [reply website/docs/terms/data-warehouse.md]: not quite sure what a post-modern data stack is...seems a little clickbait-y

File: website/docs/terms/reverse-etl.md (hunk 8 above)
Comment (KiraFuruichi) [reply website/docs/terms/reverse-etl.md:10]: yeah my b, read this wrong

File: website/docs/terms/data-warehouse.md (hunk 5 above)
Comment (johnblust) [reply website/docs/terms/data-warehouse.md]: stole this from [drew's coalesce talk](https://www.getdbt.com/coalesce-2020/the-post-modern-data-stack/), but we can toss it for now

**Comment by johnblust** [comment]
@KiraFuruichi should be good for a 👍 now

File: website/docs/terms/surrogate-key.md [hunk 11]
```diff
@@ -1,10 +1,15 @@
 ---
 id: surrogate-key
 title: Surrogate key
+description: A surrogate key is a unique identifier derived from the data itself. It's commonly a hashed value of multiple columns that will create a uniqueness id for each row.
```
Comment (KiraFuruichi) [inline website/docs/terms/surrogate-key.md]:
```suggestion
description: A surrogate key is a unique identifier derived from the data itself. It's commonly a hashed value of multiple columns that will create a unique id for each row.
```

**Review by KiraFuruichi — APPROVED** [review]
Left one final suggestion I noticed in my final review that I think we should incorporate, but other than that, it looks good to me.

**Comment by johnblust** [comment]
> Left one final suggestion I noticed in my final review that I think we should incorporate, but other than that, it looks good to me.

I changed that to ID due to a character count constraint! Does it still make sense to use ID?
